src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs

```diff
@@ -114,22 +118,36 @@ public async Task<ActionResult> UpdateOrganisationByIdAsync(Guid id, UpdateOrgan
         public async Task<ActionResult<CreateOrganisationResponseViewModel>> CreateOrganisationAsync(CreateOrganisationRequestViewModel viewModel)
         {
             if (viewModel is null)
-            {
                 throw new ArgumentNullException(nameof(viewModel));
-            }
 
-            //TODO: Task #6168
+            var result = await _createOrganisationService.CreateAsync(new CreateOrganisationRequest(
+                viewModel.OrganisationName,
+                viewModel.OdsCode,
+                viewModel.PrimaryRoleId,
+                viewModel.CatalogueAgreementSigned,
+                viewModel.Address is null ? null : new Address
+                {
+                    Line1 = viewModel.Address.Line1,
+                    Line2 = viewModel.Address.Line2,
+                    Line3 = viewModel.Address.Line3,
+                    Line4 = viewModel.Address.Line4,
+                    Town = viewModel.Address.Town,
+                    County = viewModel.Address.County,
+                    Postcode = viewModel.Address.Postcode,
+                    Country = viewModel.Address.Country
+                }
+            ));
 
-            // Canned data
-            var firstOrganisation = (await _organisationRepository.ListOrganisationsAsync()).First();
+            var response = new CreateOrganisationResponseViewModel();
 
-            var response = new CreateOrganisationResponseViewModel
+            if (!result.IsSuccess)
             {
-                Errors = null,
-                OrganisationId = firstOrganisation.OrganisationId.ToString()
-            };
+                response.Errors = result.Errors.Select(x => new ErrorMessageViewModel(x.Id, x.Field));
+                return BadRequest(response);
+            }
 
-            return Created(new Uri($"/{response.OrganisationId}", UriKind.Relative), response);
+            response.OrganisationId = result.Value;
+            return Created(new Uri($"/{result.Value}", UriKind.Relative), response);
```

Comment:
👍 I'm assuming we're leaving it as is and fixing it as part of [6259](https://buyingcatalog.visualstudio.com/Buying%20Catalogue/_sprints/taskboard/Buying%20Catalogue%20Tech%20Cell/Buying%20Catalogue/Iteration%2023) right?

Reply:
I mean I would prefer if we fixed it now so we are not introducing more tech debt & will mitigate any "copying" of the functionality. If you feel pressed to tackle this change as part of the other task then OK, but it might be a good another just to add a note to the other task 6259 to indicate the additional fix needed.

Reply:
👍 I will investigate this task as part of this. If I don't get anywhere by the end of today I'm going to merge as is - returning a 201 with the awful /{id} location header.